dtfs-central-api/src/v1/controllers/utilisation-report-service/post-add-fees-to-an-existing-payment-group.controller/index.ts

```diff
@@ -0,0 +1,68 @@
+import { ApiError } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { Response } from 'express';
+import { CustomExpressRequest } from '../../../../types/custom-express-request';
+import { InvalidPayloadError, NotFoundError } from '../../../../errors';
+import { PaymentRepo } from '../../../../repositories/payment-repo';
+import { addFeesToAnExistingPaymentGroup } from './helpers';
+import { PostAddFeesToAnExistingPaymentGroupPayload } from '../../../routes/middleware/payload-validation';
+import { FeeRecordRepo } from '../../../../repositories/fee-record-repo';
+import {
+  validateThatPaymentGroupHasFeeRecords,
+  validateThatSelectedPaymentsBelongToSamePaymentGroup,
+  validateThatSelectedPaymentsFormACompletePaymentGroup,
+} from '../../../validation/utilisation-report-service/payment-group-validator';
+
+export type PostAddFeesToAnExistingPaymentGroupRequest = CustomExpressRequest<{
+  params: {
+    reportId: string;
+  };
+  reqBody: PostAddFeesToAnExistingPaymentGroupPayload;
+}>;
+
+/**
+ * Controller for the POST fees to an existing payment group route
+ * @param req - The request object
+ * @param res - The response object
+ */
+export const postAddFeesToAnExistingPaymentGroup = async (req: PostAddFeesToAnExistingPaymentGroupRequest, res: Response) => {
+  const { reportId } = req.params;
+  const { feeRecordIds, paymentIds, user } = req.body;
+
+  try {
+    const payments = await PaymentRepo.findByIdAndReportIdWithFeeRecordsWithReportAndPayments(paymentIds, Number(reportId));
+    if (payments.length === 0) {
+      throw new NotFoundError('No payments found for the given payment IDs.');
+    }
+
+    validateThatSelectedPaymentsBelongToSamePaymentGroup(payments);
+    validateThatSelectedPaymentsFormACompletePaymentGroup(payments, paymentIds);
+
+    const existingPaymentGroupFeeRecords = payments.at(0)?.feeRecords;
+    validateThatPaymentGroupHasFeeRecords(existingPaymentGroupFeeRecords);
+
```

Comment:
I think ideally we would also repeat the validation that let's them onto this page in the first place, the status checks are covered off by the state machines, so that just leaves a check for currency matching between all of the fees payment currencies and the payments' currency.